### DISCUSSION TITLE
Downloading sub packages for analysis

### DISCUSSION THREAD

**Discussion by ghost** [materialsproject/pymatgen#2523] [Q&A] [answered]
I am a begineer to machine learning and pymatgen. I am trying to use the analysis feature on pymatgen but I keep getting an Import error.  Below is the code I am using. I have found out there are numerous subpackages that contain analysis.  I am struggling to download these packages, is there a way to download packages for analysis? 

_from pymatgen.analysis.adsorption import plot_slab
from matplotlib import pyplot as plt_

**Comment by janosh** [comment, marked as answer]
Your import

```py
from pymatgen.analysis.adsorption import plot_slab
```

should work if pymatgen was installed correctly.

You'll probably have to post more details to get help. How did you install pymatgen (`pip install pymatgen`)? What's the error message?

**Reply by ghost** [reply]
Yes I installed pymatgen using (pip install pymatgen). Below is the error message that it gives me.

ImportError                               Traceback (most recent call last)
Input In [59], in <cell line: 1>()
----> 1 from pymatgen.analysis.adsorption import plot_slab
      2 from matplotlib import pyplot as plt
      3 get_ipython().run_line_magic('matplotlib', 'inline')

File ~/opt/anaconda3/envs/py39/lib/python3.9/site-packages/pymatgen/analysis/adsorption.py:13, in <module>
     10 import os
     12 import numpy as np
---> 13 from matplotlib import patches
     14 from matplotlib.path import Path
     15 from monty.serialization import loadfn

File ~/opt/anaconda3/envs/py39/lib/python3.9/site-packages/matplotlib/__init__.py:109, in <module>
    105 from packaging.version import parse as parse_version
    107 # cbook must import matplotlib only within function
    108 # definitions, so it is safe to import from it here.
--> 109 from . import _api, _version, cbook, docstring, rcsetup
    110 from matplotlib.cbook import MatplotlibDeprecationWarning, sanitize_sequence
    111 from matplotlib.cbook import mplDeprecation  # deprecated

File ~/opt/anaconda3/envs/py39/lib/python3.9/site-packages/matplotlib/rcsetup.py:27, in <module>
     25 from matplotlib import _api, cbook
     26 from matplotlib.cbook import ls_mapper
---> 27 from matplotlib.colors import Colormap, is_color_like
     28 from matplotlib.fontconfig_pattern import parse_fontconfig_pattern
     29 from matplotlib._enums import JoinStyle, CapStyle

File ~/opt/anaconda3/envs/py39/lib/python3.9/site-packages/matplotlib/colors.py:51, in <module>
     49 from numbers import Number
     50 import re
---> 51 from PIL import Image
     52 from PIL.PngImagePlugin import PngInfo
     54 import matplotlib as mpl

File ~/opt/anaconda3/envs/py39/lib/python3.9/site-packages/PIL/Image.py:132, in <module>
    123 MAX_IMAGE_PIXELS = int(1024 * 1024 * 1024 // 4 // 3)
    126 try:
    127     # If the _imaging C module is not present, Pillow will not load.
    128     # Note that other modules should not refer to _imaging directly;
    129     # import Image and use the Image.core variable instead.
    130     # Also note that Image.core is not a publicly documented interface,
    131     # and should be considered private and subject to change.
--> 132     from . import _imaging as core
    134     if __version__ != getattr(core, "PILLOW_VERSION", None):
    135         raise ImportError(
    136             "The _imaging extension was built for another version of Pillow or PIL:\n"
    137             f"Core version: {getattr(core, 'PILLOW_VERSION', None)}\n"
    138             f"Pillow version: {__version__}"
    139         )

ImportError: dlopen(/Users/jasonpereira/opt/anaconda3/envs/py39/lib/python3.9/site-packages/PIL/_imaging.cpython-39-darwin.so, 2): Symbol not found: ____chkstk_darwin
  Referenced from: /Users/jasonpereira/opt/anaconda3/envs/py39/lib/python3.9/site-packages/PIL/.dylibs/libtiff.5.dylib (which was built for Mac OS X 11.0)
  Expected in: /usr/lib/libSystem.B.dylib
 in /Users/jasonpereira/opt/anaconda3/envs/py39/lib/python3.9/site-packages/PIL/.dylibs/libtiff.5.dylib### ###

**Reply by janosh** [reply]
Your import error is raised by PIL, not pymatgen. I would try reinstalling or installing different versions.

[This](https://github.com/python-pillow/Pillow/issues/5407) seems related.
